jerry-core/api/jerry.c

```diff
@@ -5281,6 +5281,133 @@ jerry_get_container_type (const jerry_value_t value) /**< the container object *
   return JERRY_CONTAINER_TYPE_INVALID;
 } /* jerry_get_container_type */
 
+/**
+ * Perform container operation on the given operands (add, get, set, has, delete, size, clear).
+ *
+ * @return error - if argument has an error flag or operation is unsuccessful or unsupported
+ *         result of the container operation - otherwise
+ */
+jerry_value_t
+jerry_container_operation (jerry_container_operation_t operation, /**< container operation */
+                           jerry_value_t container, /**< container */
+                           jerry_value_t *arguments, /**< list of arguments */
+                           uint32_t arguments_number) /**< number of arguments */
+{
+  jerry_assert_api_available ();
+#if ENABLED (JERRY_BUILTIN_CONTAINER)
+  if (!ecma_is_value_object (container))
+  {
+    return jerry_throw (ecma_raise_type_error (ECMA_ERR_MSG ("Container is not an object.")));
+  }
+
+  ecma_object_t *obj_p = ecma_get_object_from_value (container);
+
+  if (ecma_get_object_type (obj_p) == ECMA_OBJECT_TYPE_CLASS)
+  {
+    uint16_t type = ((ecma_extended_object_t *) obj_p)->u.class_prop.class_id;
+    ecma_extended_object_t *container_object_p = ecma_op_container_get_object (container, type);
+
+    if (container_object_p == NULL)
+    {
+      return ecma_create_error_reference_from_context ();
+    }
+
+    jerry_value_t result;
+    switch (operation)
+    {
+#if ENABLED (JERRY_BUILTIN_SET) || ENABLED (JERRY_BUILTIN_WEAKSET)
+      case JERRY_CONTAINER_OP_ADD:
+      {
+        if (arguments_number < 1 || (arguments_number >= 1 && ecma_is_value_error_reference (arguments[0])))
```

Comment:
This condition is superfluous. 
simply: `if (arguments_number < 1 || ecma_is_value_error_reference (arguments[0]))`

Same as all other cases below.